src/js/components/Marathon.jsx

```diff
@@ -44,21 +46,24 @@ var Marathon = React.createClass({
   },
 
   componentDidMount: function () {
-    this.onRouteChange();
+    PluginActions.requestPlugins();
 
+    this.onRouteChange();
     this.bindKeyboardShortcuts();
 
-    PluginStore.once(PluginEvents.BOOTSTRAP_COMPLETE, () => {
-      this.startPolling();
-    });
-
-    PluginStore.bootstrap();
+    PluginStore.on(PluginEvents.CHANGE, this.onPluginStoreChange);
   },
 
   componentWillReceiveProps: function () {
     this.onRouteChange();
   },
 
+  onPluginStoreChange: function () {
+    if (PluginStore.getPluginLoadingState() > States.STATE_LOADING) {
```

Comment:
This is an unconfident condition. What if there will be an error-state introduced in the future, that is greater then STATE_LOADING?

Why did you removed the `PluginEvents.BOOTSTRAP_COMPLETE` handler?